### PR TITLE
Page/add proposal

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
- refactor form input for multi useage
- add an AddProposal page

### CODE REVIEW

File: src/app/components/FormInput/FormInput.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+`
+const Input = styled.input`
+  width: 100%;
+  background-color: transparent;
+  border: 1px solid rgba(0, 0, 0, 0.5);
+  padding: 8px 12px;
+  margin-bottom: 20px;
+  border-radius: 12px;
+  font-weight: bold;
+  font-size: 1.125rem;
```
Comment (PReichetanz) [inline src/app/components/FormInput/FormInput.tsx:61]: Why using rem here but px before and after?
```suggestion
  font-size: 18px;
```

File: src/app/components/FormInput/FormInput.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      rgba(0, 0, 0, 0.3) 0px 18px 36px -18px inset;
+  }
+`
+const TextArea = styled.textarea`
+  width: 100%;
+  background-color: transparent;
+  border: 1px solid rgba(0, 0, 0, 0.5);
+  border-radius: 12px;
+  resize: none;
+  font-size: 1rem;
```
Comment (PReichetanz) [inline src/app/components/FormInput/FormInput.tsx:77]:
```suggestion
  font-size: 16px;
```

File: src/app/components/FormInput/FormInput.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      rgba(0, 0, 0, 0.3) 0px 18px 36px -18px inset;
+  }
+`
+const InputWrapper = styled.div`
+  position: relative;
+`
+
+const TextLimitDisplay = styled.span<{ diff: number }>`
+  position: absolute;
+  bottom: 1.3rem;
```
Comment (PReichetanz) [inline src/app/components/FormInput/FormInput.tsx:95]:
```suggestion
  bottom: 20.8px;
```

File: src/app/components/FormInput/FormInput.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  right: 10px;
+  font-size: 0.8rem;
```
Comment (PReichetanz) [inline src/app/components/FormInput/FormInput.tsx:97]:
```suggestion
  font-size: 12.8px;
```

File: src/app/components/TopicForm/TopicForm.tsx [hunk 5]
```diff
@@ -41,36 +42,27 @@ export default function TopicForm({
...
-      </InputWrapper>
-
+      <FormInput
+        type="text"
+        name="title"
+        required
+        diff={titleDiff}
+        value={title}
+        onChange={event =>
+          event.target.value.length <= MAX_TITLE_LENGTH &&
```
Comment (PReichetanz) [inline src/app/components/TopicForm/TopicForm.tsx:52]: What the hell are these => <= doing? o.O

File: src/app/pages/AddProposal/AddProposal.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  color: var(--c-gray-50);
+  text-decoration: none;
+  align-content: center;
+`
+
+const Header = styled.h1`
+  color: var(--c-gray-50);
+  margin: 30px;
+  margin-top: 20vh;
+  font-size: 4rem;
```
Comment (PReichetanz) [inline src/app/pages/AddProposal/AddProposal.tsx:112]:
```suggestion
  font-size: 64px;
```

**Review by PReichetanz — COMMENTED** [review]
Looks good to me, only minor suggestions on changing rem to px and a question about this strange  => <= thing :D

File: src/app/components/FormInput/FormInput.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import styled from 'styled-components'
+
+type FormInputProps = {
+  type: 'text' | 'textArea'
+  value: string
+  name: string
+  diff?: number
```
Comment (lajanzen) [inline src/app/components/FormInput/FormInput.tsx:8]: It took me a while to figure out what `diff` is, maybe think about renaming it? 
What about `charactersLeft` for example?

File: src/app/components/TopicForm/TopicForm.tsx (hunk 5 above)
Comment (lajanzen) [reply src/app/components/TopicForm/TopicForm.tsx:52]: I think it's the "kleiner gleich" sign, but I was also a little confused at first 😅

File: src/app/pages/AddProposal/AddProposal.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+export default function AddProposal({
+  onSubmit,
+  topics,
+}: AddProposalProps): JSX.Element {
+  const [description, setDescription] = useState('')
+  const descriptionDiff = MAX_DESCRIPTION_LENGTH - description.length
+
+  const params = useParams()
+  console.log(topics)
```
Comment (lajanzen) [inline src/app/pages/AddProposal/AddProposal.tsx:24]: Do you need this? 

```suggestion
```

File: src/app/pages/AddProposal/AddProposal.tsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+  const topic = topics.find(topic => topic.id === params.topicId)
+  const needs = topic ? topic.needs : []
+
+  const nav = useNavigate()
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    onSubmit()
+  }
```
Comment (lajanzen) [inline src/app/pages/AddProposal/AddProposal.tsx:34]: Not sure if this is necessary. I think in line 48, you could also just apply `onSubmit={onSubmit}`.

**Pull request merged**